lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -757,24 +646,19 @@ bool DeclVisitor::VisitFunctionDecl(const clang::FunctionDecl *FD) {
           return false;
         }
 
-        NamedTypedRegister ParamReg(LocationID);
-        ParamReg.Type() = *ParamType;
-        ArgumentsInserter.insert(ParamReg);
+        NamedTypedRegister &ParamReg = ArgumentsInserter.emplace(LocationID);
+        ParamReg.Type() = std::move(ParamType);
       }
     }
   }
 
   // Update the name if in the case it got changed.
-  auto &ModelFunction = Model->Functions()[Function->Entry()];
+  auto &ModelFunction = Model->Functions()[std::get<0>(*Function)];
   setCustomName(ModelFunction, FD->getName());
 
-  // Clone the other stuff.
-  ModelFunction.OriginalName() = Function->OriginalName();
-  ModelFunction.ExportedNames() = Function->ExportedNames();
-
```

Comment:
Why?

Reply:
Ping

Reply:
Ah, right, this one is weird. I'm not 100 sure why we're even doing this! We're copying data from the function, key for which we have received in the constructor and _never_ modified into the very same function, but after looking it up in the model.

Reply:
@dwarfyresearch 

1. Can you comment on this?
```c++
  ModelFunction.OriginalName() = Function->OriginalName();
  ModelFunction.ExportedNames() = Function->ExportedNames();
```
2. I also see:
```c++
void HeaderToModel::HandleTranslationUnit(ASTContext &Context) {
  Model->Architecture() = Model->Architecture();
  Model->DefaultABI() = Model->DefaultABI();
```
3. It seems like `HeaderToModel` (and `DeclVisitor`) constructor should take a `model::Function::Key`, i.e., a `MetaAddress`.
   Alterantively, we can keep a reference to the `model::Function` but then there's no point in looking it up as we're doing here: `auto &ModelFunction = Model->Functions()[Function->Entry()];`.

Reply:
1. I think it is the code that ensures that we do update the function name in the Model if it got renamed by the user. If this code/feature should not be covering that (so the user cannot rename the function with this feature), we can remove this code.
2. Can be removed. Dead code.
3. >It seems like HeaderToModel (and DeclVisitor) constructor should take a model::Function::Key, i.e., a MetaAddress.

+1

Reply:
Done.